lib/chromium.js

```diff
@@ -182,11 +216,21 @@ module.exports = function (b, opts) {
         });
       }
     });
-    server.listen(serverPort, function (err) {
+    allocatePort(serverPort, function (err, port) {
       if (err) {
         b.emit('error', err);
         finish();
+        return;
+      }
+      if (!opts.url) {
+        opts.url = 'https://localhost:' + port;
```

Comment:
Can adding this asynchronously create some race condition that I am not aware of?